Stop users from giving themselves karma

himmelmaus opened the pull request:
Similar to previous PR, test name with "~d" at the end - except should pass the linter this time

himmelmaus committed:
Update karma.py

actually might pass this time

neanias commented:
For future reference, you can add and push another commit and the CI will auto-build it again.

neanias commented on ircbot/plugins/karma.py:
```diff
@@ -72,7 +72,7 @@ def message_hook(bot, channel, sender, message):
         reason = m.group('reason')
         term = term.strip().lower()
-        if term == sender.lower():
+        if term == sender.lower() or term + "~d" == sender.lower():
```
The CI hasn't run yet, but I'm not sure how the + associates.

neanias commented:
@findoslice, have you tested this?

himmelmaus commented:
@neanias tested as to whether it works as a patch or with the linter? If the former, I have a bot with the similar karma configuration and on a shared discord/irc channel where it works so to my knowledge it will.

neanias commented:
I was wondering whether the (+) has higher precedence than (==)

himmelmaus commented:
Oh, in that case it definitely does (tested that myself right now to be sure)

neanias commented:
Fair play then. See follow up discussion on IRC

neanias commented:
@hreeder Got any opinions?

hreeder commented:
I'd probably like to see the suffix moved to the config file, so it's something we can expand in future (say we have to recognize users from another service that come in with ~s on the end?)

In that event, maybe it becomes a regex where we insert `term` dynamically? Open to ideas on how to achieve that.

hreeder commented:
Thinking about it, if we went down that route of having some kind of configurable regex, we could drop the or entirely, because we could perform that or in the regex

neanias commented:
Regex++

> On 17 Mar 2018, at 22:41, Harry Reeder <notifications@github.com> wrote:
> 
> Thinking about it, if we went down that route of having some kind of configurable regex, we could drop the or entirely, because we could perform that or in the regex
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

himmelmaus committed:
Update self karma prevention to use regex

Uses re.fullmatch with "~d", "~s", or nothing; instead of appending "~d" and testing equality.

himmelmaus committed:
Update regex self karma prevention to actually work

Made a dumb mistake and tested if term == term (pls don't laugh at me)

hreeder commented:
Looking better, @findoslice - now I think just move it out to `bot.config` (maybe `bot.config['Karma']['self_re']`)

himmelmaus committed:
add whitespace to keep the linter happy

no changes to logic, had to add a whitespace so the code would pass the linter

himmelmaus committed:
add karma regex to config file

himmelmaus committed:
Move Karma regex to config

hreeder commented:
LGTM :+1: 

Thanks @findoslice
